### PR TITLE
Task 1, 2, 3, 4

### PR DESCRIPTION
**Pull request by drupali09**

### CODE REVIEW

**Comment by drupali09** [comment]
Task 5

File: resources/database.sql [hunk 1]
```diff
@@ -9,6 +9,7 @@ CREATE TABLE todos (
   id INT(11) NOT NULL AUTO_INCREMENT,
   user_id INT(11) NOT NULL,
   description VARCHAR(255),
+  is_finished BOOLEAN,
```
Comment (T0RAT0RA) [inline resources/database.sql]: How would we migrate an existing database?

File: src/controllers.php [hunk 2]
```diff
@@ -54,36 +57,101 @@
...
+})->bind('homepage');
+
 $app->post('/todo/add', function (Request $request) use ($app) {
     if (null === $user = $app['session']->get('user')) {
         return $app->redirect('/login');
     }
 
     $user_id = $user['id'];
     $description = $request->get('description');
+  if ($description == ''){
```
Comment (T0RAT0RA) [inline src/controllers.php]: `$description` is not trimmed.

File: src/controllers.php [hunk 3]
```diff
@@ -54,36 +57,101 @@
...
+    $request->getSession()
+    ->getFlashBag()
+    ->add('success', 'Your todo has been deleted.');
+
     return $app->redirect('/todo');
+});
+
+$app->match('/todo/done/{id}', function (Request $request , $id) use ($app) {
+
+  $sql = "UPDATE todos SET is_finished = 1 WHERE id = '$id'";
```
Comment (T0RAT0RA) [inline src/controllers.php]: Security breach: user can complete any `todo` in the system.

File: config/config.yml.dist [hunk 4]
```diff
@@ -1,5 +1,5 @@
 database:
     host    : localhost
-    dbname  : ac_todos
+    dbname  : sym
```
Comment (T0RAT0RA) [inline config/config.yml.dist]: Why the db name changed?

File: src/controllers.php [hunk 5]
```diff
@@ -54,36 +57,101 @@
             'todo' => $todo,
         ]);
     } else {
-        $sql = "SELECT * FROM todos WHERE user_id = '${user['id']}'";
+        $sql = "SELECT * FROM todos WHERE is_finished = 0 AND user_id = '${user['id']}'";
```
Comment (T0RAT0RA) [inline src/controllers.php]: Why filtering out completed `todos`?
How do a user see all his `todos`?

File: composer.json [hunk 6]
```diff
@@ -22,7 +22,10 @@
         "symfony/twig-bridge": "~2.3",
         "symfony/validator": "~2.3",
         "doctrine/dbal": "~2.2",
-        "deralex/yaml-config-service-provider": "1.0.x-dev"
+        "deralex/yaml-config-service-provider": "1.0.x-dev",
+        "franmomu/silex-pagerfanta-provider": "1.0.*@dev",
+        "pagerfanta/pagerfanta": "dev-master",
+        "fzaninotto/faker": "1.1.*@dev"
```
Comment (T0RAT0RA) [inline composer.json]: I don't see where this dependency is used.

File: resources/fixtures.sql [hunk 7]
```diff
@@ -11,4 +11,5 @@ INSERT INTO todos (user_id, description) VALUES
 (1, 'Accumsan nunc vitae'),
 (2, 'Lorem ipsum'),
 (2, 'In lacinia est'),
-(2, 'Odio varius gravida');
\ No newline at end of file
+(2, 'Odio varius gravida');
+(2, 'My todo', 1);
```
Comment (T0RAT0RA) [inline resources/fixtures.sql]: This can't work because we insert only 2 fields: `(user_id, description)`

**Pull request closed without merging**